test/functional/tests/failover_standby/test_failover_advanced.py

```diff
@@ -52,19 +122,220 @@ def test_failover_during_background_cleaning(pyocf_ctx, cache_mode, cls, cleanin
+            primary_node
+        ):
+            checksum1 = TestRun.executor.run(f"md5sum {core.path}").stdout.split()[0]
+
+        with TestRun.step(
+            f"Switch back to the {cache_mode} cache mode without flush"
+        ), TestRun.use_dut(primary_node):
+            primary_node.cache.set_cache_mode(cache_mode, flush=False)
+
+        with TestRun.step(f"Switch to {cleaning_policy} cleaning policy"), TestRun.use_dut(
+            primary_node
+        ):
+            primary_node.cache.set_cleaning_policy(cleaning_policy)
+
+            if cleaning_policy == CleaningPolicy.alru:
+                TestRun.LOGGER.info("Configure ALRU to trigger immediately\n")
+                params = FlushParametersAlru(
+                    activity_threshold=Time(milliseconds=0),
+                    wake_up_time=Time(seconds=0),
+                    staleness_time=Time(seconds=1),
+                )
+                primary_node.cache.set_params_alru(params)
+
+        with TestRun.step("Wait 2s"):
+            time.sleep(2)
+
+        with TestRun.step(
+            "Verify cleaner is progressing by inspecting dirty statistics"
+        ), TestRun.use_dut(primary_node):
+            dirty2 = primary_node.cache.get_statistics(percentage_val=True).usage_stats.dirty
+            TestRun.LOGGER.info(f"Dirty stats change: {dirty}% -> {dirty2}%")
+
+            # make sure there is cleaning progress
+            if dirty2 >= dirty:
+                TestRun.LOGGER.error("No cleaning progress detected")
+
+            # make sure there is dirty data left to clean
+            if dirty2 < 20:
+                TestRun.LOGGER.error("Not enough dirty data")
+
+        with TestRun.step(f"Power off the main DUT"), TestRun.use_dut(primary_node):
+            timed_async_power_cycle()
+
+        with TestRun.step("Stop cache DRBD on the secondary node"), TestRun.use_dut(secondary_node):
+            secondary_node.cache_drbd.down()
+
+        with TestRun.step("Set backup DUT as primary for core DRBD"), TestRun.use_dut(
+            secondary_node
+        ):
+            secondary_node.core_drbd.set_primary()
+
+        with TestRun.step("Deatch cache drive from standby cache instance"), TestRun.use_dut(
+            secondary_node
+        ):
+            secondary_node.cache.standby_detach()
+
+        with TestRun.step(
+            "Activate standby cache instance directly on the cache drive"
+        ), TestRun.use_dut(secondary_node):
+            secondary_node.cache.standby_activate(secondary_node.cache_dev)
+
+        with TestRun.step("Verify there is some dirty data after failover"), TestRun.use_dut(
+            secondary_node
+        ):
+            dirty3 = secondary_node.cache.get_statistics(percentage_val=True).usage_stats.dirty
+            if dirty3 > dirty2:
+                TestRun.LOGGER.error("Unexpeted increase in dirty cacheline count")
+            elif dirty3 == 0:
+                TestRun.LOGGER.error(
```

Comment:
I'm not sure. Error here might indicate problem in CAS, although it is not the purpose of this test to check exactly this. My understanding is that exception would be the kind of error that is clearly not caused by the tested product, e.g. failure to create partitions.

Reply:
I see it as a way to notify the person that will be analyzing the results that something went wrong with the test in a way that it was not able to perform the actual testing. I consider such a feedback useful as it may help to decide such a person whether to rerun this test or report the problem as a bug, especially if the person is not CAS expert. If I were the author of this test I would use `TestRun.LOGGER.exception()`, `pytest.skip()` or at least I would add some additional info in the error message.

Reply:
ok, let's do it